src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs

```diff
@@ -34,15 +35,18 @@ internal partial class VisualStudioWorkspaceDiagnosticAnalyzerProviderService :
         private static readonly AnalyzerAssemblyLoader s_analyzerAssemblyLoader = new AnalyzerAssemblyLoader();
 
         [ImportingConstructor]
-        public VisualStudioWorkspaceDiagnosticAnalyzerProviderService(VisualStudioWorkspaceImpl workspace)
+        public VisualStudioWorkspaceDiagnosticAnalyzerProviderService(
+            [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
         {
+            var dte = (EnvDTE.DTE)serviceProvider.GetService(typeof(EnvDTE.DTE));
```

Comment:
@jasonmalinowski, I'm confused how this can fix the free-threading problems when you're still calling UI-bound functions from the constructor of MEF components.